config/src/config/partial.rs

```diff
@@ -102,10 +102,23 @@ pub struct ConsensusConstants {
     /// Timestamp at checkpoint 0 (the start of epoch 0)
     #[serde(default)]
     pub checkpoint_zero_timestamp: Option<i64>,
+
     /// Seconds between the start of an epoch and the start of the next one
     #[serde(default)]
-    #[serde(rename = "checkpoint_period_seconds")]
-    pub checkpoint_period: Option<u16>,
+    #[serde(rename = "checkpoints_period_seconds")]
+    pub checkpoints_period: Option<u16>,
+
+    /// Genesis block hash value
+    #[serde(default)]
+    pub genesis_hash: Option<Vec<u8>>,
+
+    /// Decay value for reputation demurrage function
+    #[serde(default)]
+    pub reputation_demurrage: Option<f64>,
```

Comment:
@witnet/devs Floating-point values are not suitable for consensus-critical constants because they are not deterministic. They can arbitrarily trade off accuracy for precision, and they can be rounded up or down differently on distinct architectures, processors, operating systems and specific devices. Many [FPU]s even assimilate [subnormal values][denormal] to zero.

To put this into perspective, this is the very reason why the Ethereum Virtual Machine does not support float types either.

Even though Rust complies with IEEE 754, it still behaves inconsistently when dealing with floating-point values across different platforms. This has been [discussed at length by the community][broken-floats], but no final RFC has been adopted.

We will be better off using [fixed-point arithmetics][fixed-point], as we can compromise range and even precision for the sake of accuracy and determinism. A bunch of good-looking fixed-point arithmetic libraries are just [one Cargo search away][cargo].

[FPU]: https://en.wikipedia.org/wiki/Floating-point_unit
[denormal]: https://en.wikipedia.org/wiki/Denormal_number
[broken-floats]: https://internals.rust-lang.org/t/pre-rfc-dealing-with-broken-floating-point/2673
[fixed-point]: https://en.wikipedia.org/wiki/Fixed-point_arithmetic
[cargo]: https://crates.io/search?q=fixed%20point

Reply:
I have been checking fixed-point arithmetic libraries and currently they do not implement deserialize traits. Due to demurrage algorithm it is not necessary to Testnet, I would add an issue instead to solve now. What do you think?

Reply:
I agree, we can defer the decision on this for the time being. Issue #172 has been opened to track progress on assessing, presenting, discussing, deciding and documenting our strategy regarding consensus-critical real numbers.